clang/test/SemaHLSL/BuiltIns/Dot2Add-errors.hlsl

```diff
@@ -0,0 +1,11 @@
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -emit-llvm-only -disable-llvm-passes -verify
+
+bool test_too_few_arg() {
+  return __builtin_hlsl_dot2add();
+  // expected-error@-1 {{too few arguments to function call, expected 3, have 0}}
+}
+
+bool test_too_many_arg(half2 p1, half2 p2, float p3) {
+  return __builtin_hlsl_dot2add(p1, p2, p3, p1);
+  // expected-error@-1 {{too many arguments to function call, expected 3, have 4}}
+}

```

Comment:
here are a few things that I would  expect to fail that should be included in this file
```hlsl
float fn(half2 p1, float2 p2, float p3) {...} // arg 2 needs to be a half2 type
float fn(float2 p1, half2 p2, float p3) {...} // arg 1 needs to be a half2 type
float fn(float2 p1, float2 p2, float p3) {...} // arg 1 & 2 needs to be a half2 type
float fn(float2 p1, float2 p2, double p3) {...} // arg 3 needs to be a float
```